### PR TITLE
Autoloader Plugin

### PR DESCRIPTION
**Pull request by Golmote**
Discussion about this lies in #456. Our main issue was to determine the dependencies for each language. Here, with a map automatically kept up to date with a gulp task, the issue is solved.

See [online example here](http://golmote.free.fr/prism-autoloader-new/plugins/autoloader/index.html).

What do you think?

### CODE REVIEW

File: plugins/autoloader/prism-autoloader.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
+(function () {
+	if (typeof self === 'undefined' || !self.Prism || !self.document || !document.createElement) {
+		return;
+	}
+
+	// The dependencies map is built automatically with gulp
+	var lang_dependencies = /*languages_placeholder[*/{"javascript":"clike","actionscript":"javascript","aspnet":"markup","bash":"clike","c":"clike","csharp":"clike","cpp":"c","coffeescript":"javascript","css-extras":"css","d":"clike","dart":"clike","fsharp":"clike","glsl":"clike","go":"clike","groovy":"clike","haml":"ruby","handlebars":"markup","jade":"javascript","java":"clike","less":"css","markdown":"markup","objectivec":"c","php":"clike","php-extras":"php","processing":"clike","qore":["clike"],"jsx":["markup","javascript"],"ruby":"clike","sass":"css","scss":"css","scala":"java","smarty":"markup","swift":"clike","textile":"markup","twig":"markup","typescript":"javascript","wiki":"markup"}/*]*/;
```
Comment (mAAdhaTTah) [inline plugins/autoloader/prism-autoloader.js]: Is this an artifact of the gulp task?

**Comment by mAAdhaTTah** [comment]
Awesome, I'll have to play around with this.

File: plugins/autoloader/prism-autoloader.js (hunk 1 above)
Comment (Golmote) [inline plugins/autoloader/prism-autoloader.js]: I'm not sure of what you mean by artifact, but yes, it was built automatically through gulp.

**Comment by Golmote** [comment]
@mAAdhaTTah: Please let me know if you find any flaw.

File: plugins/autoloader/prism-autoloader.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	 * Load a grammar with its dependencies
+	 * @param {string} lang
+	 * @param {function=} success
+	 * @param {function=} error
+	 */
+	var loadLanguage = function(lang, success, error) {
+		var load = function() {
+			var force = false;
+			// Do we want to force reload the grammar?
+			if(lang.indexOf('!') > 0) {
```
Comment (Golmote) [inline plugins/autoloader/prism-autoloader.js]: TODO: Change to `>=`

File: plugins/autoloader/prism-autoloader.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		};
+		s.onerror = function() {
+			document.body.removeChild(s);
+			error && error();
+		};
+		document.body.appendChild(s);
+	};
+
+	/**
+	 *
```
Comment (Golmote) [inline plugins/autoloader/prism-autoloader.js]: TODO: Returns the path to a grammar, using the language_path and use_minified config keys.

File: plugins/autoloader/prism-autoloader.js (hunk 1 above)
Comment (mAAdhaTTah) [inline plugins/autoloader/prism-autoloader.js]: The comment strings at the beginning and end.

Comment (Golmote) [inline plugins/autoloader/prism-autoloader.js]: Oh! Yes, the gulp task use them to know where to insert the generated object.

File: gulpfile.js [hunk 4]
```diff
@@ -66,18 +68,41 @@ gulp.task('show-language-plugin', function (cb) {
...
+					if(cpt === l) {
+						cb && cb();
+					}
+				};
+
+				tasks.forEach(function(task) {
+					var stream = gulp.src(task.plugin)
+						.pipe(replace(
+							/\/\*languages_placeholder\[\*\/[\s\S]*?\/\*\]\*\//,
+							'/*languages_placeholder[*/' + task.map + '/*]*/'
```
Comment (Golmote) [inline gulpfile.js]: @mAAdhaTTah: Here is where the magic happens.

Comment (mAAdhaTTah) [inline gulpfile.js]: Yeah, I was just wondering if the comment needed to be included in the final product. It's not a huge deal :).

Comment (Golmote) [inline gulpfile.js]: Well, it is discarded in the minified version of the plugin :D

**Pull request merged**